pkg/controller/cassandracluster/pod_operation.go

```diff
@@ -290,46 +299,43 @@ func (rcc *ReconcileCassandraCluster) ensureDecommission(cc *api.CassandraCluste
 
 	if podLastOperation.Name != api.OperationDecommission {
 		logrus.WithFields(logrus.Fields{"cluster": cc.Name, "rack": dcRackName,
-			"lastOperation": podLastOperation.Name}).Warnf("We should decommission only if pod.Operation == decommission, not the case here")
+			"lastOperation": podLastOperation.Name}).Warnf("There is another operation than decommission that was asked")
 		return continueResyncLoop, nil
 	}
 
 	switch podLastOperation.Status {
 
-	case api.StatusToDo:
+	case api.StatusToDo, api.StatusContinue:
 
 		return rcc.ensureDecommissionToDo(cc, dcName, rackName, status)
 
-	case api.StatusOngoing,
-		api.StatusFinalizing:
-
-		if podLastOperation.Pods == nil || podLastOperation.Pods[0] == "" {
-			return breakResyncLoop, fmt.Errorf("For Status Ongoing we should have a PodLastOperation Pods item")
-		}
+	case api.StatusFinalizing:
 		lastPod, err := rcc.GetPod(cc.Namespace, podLastOperation.Pods[0])
 		if err != nil {
-			if !apierrors.IsNotFound(err) {
-				return breakResyncLoop, fmt.Errorf("failed to get last cassandra's pods '%s': %v",
-					podLastOperation.Pods[0], err)
+			if apierrors.IsNotFound(err) {
+				return rcc.deletePodPVC(cc, dcName, rackName, status, lastPod)
```

Comment:
is that what we want to do ? 
Was there a reason that we didn't remove it before ?

Reply:
I just changed when it's done to fix the problem